plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseTablespace.java

```diff
@@ -0,0 +1,498 @@
+        DISK_SYSTEM_DATA(3),
+        DISK_USER_DATA(4),
+        DISK_SYSTEM_TEMP(5),
+        DISK_USER_TEMP(6),
+        DISK_SYSTEM_UNDO(7),
+        VOLATILE_USER_DATA(8),
+        UNKNOWN(-1);
+        
+        private int stateIdx;
+        
+        private TbsType(int stateIdx) {
+            this.stateIdx = stateIdx;
+        }
+        
+        /**
+         * Get TBS type get its index value. 
+         */
+        public static TbsType getTbsTypeByIdx(int stateIdx) {
+            for (TbsType type : TbsType.values()) {
+                if (stateIdx == type.stateIdx) {
+                    return type;
+                }
+            }
+            
+            return UNKNOWN;
+        }
+    }
+    
+    public enum State {
+        OFFLINE(1),
+        ONLINE(2),
+        BACKUP_OFFLINE(5),
+        BACKUP_ONLINE(6),
+        DROPPPED(128),
+        DISCARDED(1024),
+        BACKUP_DISCARDED(1028),
+        UNKNOWN(-1);
+        
+        private int stateIdx;
+        
+        private State(int stateIdx) {
+            this.stateIdx = stateIdx;
+        }
+        
+        /**
+         * Get TBS status by its state index value. 
+         */
+        public static State getStateByIdx(int stateIdx) {
+            for (State status : State.values()) {
+                if (stateIdx == status.stateIdx) {
+                    return status;
+                }
+            }
+            
+            return UNKNOWN;
+        }
+    }
+    
+    private int id;
+    private String name;
+    private String extentManagement;
+    private String segmentManagement;
+    private int dataFileCount;
+    private boolean isLogCompression;
+    private int pageSizeInBytes;
+    
+    private TbsType tbsType;
+    private State state;
+    
+    private volatile Long availableSize;
+    private volatile Long usedSize;
+    private String qry4Size;
+    
+    final FileCache fileCache = new FileCache();
+    
+    protected AltibaseTablespace(AltibaseDataSource dataSource, ResultSet dbResult) {
+        super(dataSource, true);
+        
+        this.id                 = JDBCUtils.safeGetInt(dbResult, "ID");
+        this.name               = JDBCUtils.safeGetString(dbResult, "NAME");
+        this.tbsType            = TbsType.getTbsTypeByIdx(JDBCUtils.safeGetInt(dbResult, "TYPE"));
+        this.state              = State.getStateByIdx(JDBCUtils.safeGetInt(dbResult, "STATE"));
+        this.extentManagement   = JDBCUtils.safeGetString(dbResult, "EXTENT_MANAGEMENT");
+        this.segmentManagement  = JDBCUtils.safeGetString(dbResult, "SEGMENT_MANAGEMENT");    
+        this.dataFileCount      = JDBCUtils.safeGetInt(dbResult, "DATAFILE_COUNT");
+        this.pageSizeInBytes    = JDBCUtils.safeGetInt(dbResult, "PAGE_SIZE");
+        this.isLogCompression   = (JDBCUtils.safeGetInt(dbResult, "ATTR_LOG_COMPRESS") == 1);
+        
+        setQry4Size();
+    }
+
+    /* Reference: https://docs.altibase.com/x/j4KZ */

```

Comment:
Unfortunatelly, I can not accept a link to the non-English documentation here. 
But, maybe I didn't find the localization button on the web-site?

Reply:
Unfortunately, they have not translated the target page into in English. I'll remove the comment.